src/main/java/org/infinispan/tutorial/db/LocationWeatherMarshallingContext.java

```diff
@@ -2,44 +2,23 @@
 
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
-import org.infinispan.client.hotrod.marshall.MarshallerUtil;
-import org.infinispan.protostream.SerializationContext;
-import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
-import org.infinispan.tutorial.data.LocationWeather;
+import org.infinispan.protostream.GeneratedSchema;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
 
-import java.io.IOException;
 import java.util.Objects;
 
-import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
-
 public final class LocationWeatherMarshallingContext {
 
-    public static void initSerializationContext(RemoteCacheManager cacheManager) {
-        System.out.println("---- LocationWeatherMarshallingContext - initialize the serialization context for LocationWeather class ----");
-        Objects.requireNonNull(cacheManager);
-
-        // Get the serialization context of the client
-        SerializationContext ctx = MarshallerUtil.getSerializationContext(cacheManager);
-
-        // Use ProtoSchemaBuilder to define a Protobuf schema on the client
-        ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
-        String fileName = "weather.proto";
-        String protoFile = null;
-        try {
-            protoFile = protoSchemaBuilder
-                    .fileName(fileName)
-                    .addClass(LocationWeather.class)
-                    .packageName("org.infinispan.tutorial.data")
-                    .build(ctx);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+   public static void initSerializationContext(RemoteCacheManager cacheManager) {
+      System.out.println("---- LocationWeatherMarshallingContext - initialize the serialization context for LocationWeather class ----");
+      Objects.requireNonNull(cacheManager);
 
-        // Retrieve metadata cache
-        RemoteCache<String, String> metadataCache =
-                cacheManager.getCache(PROTOBUF_METADATA_CACHE_NAME);
+      // Retrieve metadata cache
+      RemoteCache<String, String> metadataCache =
+            cacheManager.getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
 
-        // Define the new schema on the server too
-        metadataCache.put(fileName, protoFile);
-    }
+      GeneratedSchema schema = new LocationWeatherSchemaImpl();
+      // Define the new schema on the server too
+      metadataCache.put(schema.getProtoFileName(), schema.getProtoFile());
```

Comment:
so now we have the schema registered on the server but I do not see where it is needed and registered on the client.